### DISCUSSION TITLE
Unsealed RelayCommand

### DISCUSSION THREAD

**Discussion by czmirek** [CommunityToolkit/dotnet#694] [Ideas]
Hello,

I'm building an Android app with .MAUI with .NET 7 and I wanted to try the behavior of `FlowExceptionsToTaskScheduler` boolean inside the `RelayCommandAttribute` which I've just discovered today.

Few months back I couldn't find a way to fire `TaskScheduler.UnobservedTaskException` or `Android.Runtime.AndroidEnvironment.UnhandledExceptionRaiser`. Any exception just crashes the application, only my own `try...catch` blocks can prevent that.

But now setting `FlowExceptionsToTaskScheduler = true` works just great and the exception goes to `UnobservedTaskException`.

```csharp
    [RelayCommand] //crashes the app, does not go to UnobservedTaskException
    async Task CrashingMethod1()
    {
        throw new Exception();
    }

    [RelayCommand(FlowExceptionsToTaskScheduler = true)] //goes to UnobservedTaskException event
    async Task CrashingMethod2()
    {
        throw new Exception();
    }
```

Excellent! 

`FlowExceptionsToTaskScheduler` is false by default...but I want it to be `true` by default everywhere, I really need to avoid crashing of the app at all costs, showing some error dialogs instead.

Having `[RelayCommand(FlowExceptionsToTaskScheduler = true)]` everywhere works...but it's kinda ugly.

So let's just make my own class, something like `GuardedRelayCommandAttribute` which extends the `RelayCommandAttribute` and sets the property to `true` by default on initialization.

...except I cannot. Because `RelayCommandAttribute` is sealed.

Thoughts?

**Comment by kmgallahan** [comment]
You are using `RelayCommandAttribute`, which uses Roslyn source generation to create `AsyncRelayCommand`s.

Even if it were unsealed so you could use inheritance, it would no longer be picked up by the source generation logic.

Instead, you should propose that a configuration option should be provided to set the default value used during source generation.
